rasa/core/nlg/response.py

```diff
@@ -25,28 +25,58 @@ def __init__(self, responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
         """
         self.responses = responses
 
+    def _matches_filled_slots(
+        self, filled_slots: Dict[Text, Any], response: Dict[Text, Any],
+    ) -> bool:
+        """Checks if the conditional response variation matches the filled slots."""
+        satisfied_constraints = []
+        constraints = response.get("condition")
+        for constraint in constraints:
+            name = constraint["name"]
+            value = constraint["value"]
+            if filled_slots.get(name) == value:
+                satisfied_constraints.append(constraint)
+
+        if len(satisfied_constraints) == len(constraints):
+            return True
+
+        return False
```

Comment:
optional: slightly simpler in my opinion 
```suggestion
        constraints = response.get("condition")
        for constraint in constraints:
            name = constraint["name"]
            value = constraint["value"]
            if filled_slots.get(name) != value:
                return False

        return True
```

Reply:
Very nice! I will follow this suggestion 👍